components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt

```diff
@@ -240,4 +241,16 @@ class GeckoEngineView @JvmOverloads constructor(
     companion object {
         internal const val DARK_COVER = 0xFF2A2A2E.toInt()
     }
+
+    override fun dispatchKeyEvent(event: KeyEvent): Boolean {
+        if (event.isCtrlPressed) {
+            when (event.getKeyCode()) {
+                KeyEvent.KEYCODE_R -> {
+                    currentSession?.reload()
+                    return true
+                }
+            }
+        }
+        return super.dispatchKeyEvent(event)
+    }
```

Comment:
The approach of intercepting the keys here is right. But I would like us to find a solution to not have this implemented hard-coded in every `engine` implementation and instead have the right hooks for keys here and an independent component hat can implement those for all engines. Does this make sense?

Reply:
I mentioned this in my initial comment but I need guidance on what that extensible implementation would look like.  I understand that this should be sending these events to the embedding application so it can call things like components.useCases.sessionUseCases.reload() but I'm not sure where to start.  Should `GeckoEngineView` take an interface with methods for actions like reload?  Should it pass along the `KeyEvent` themselves?  Is there some other approach?

Reply:
(Not an android-components dev, so take with a grain of salt).
I'd expect an interface with a single method (`dispatchKeyEvent` maybe?, `handleKeyEvent`?) returning a boolean for whether the event was handled.
There would then be an implementation of the class checking for the hotkeys and handling them appropriately.

Reply:
I can define a `KeyEventHandler` interface.  Should I add a `setKeyEventHandler` method to `GeckoEngineView` or are these injected in some other way?  Should only Fenix implement the new interface defined in android-components or do you envision some code reuse with the reference browser?  I am primarily interested in the former.

Reply:
I think this starts to go into the right direction. Ideally we would have some kind of handler that can be set on an `EngineView` (any implementation!) and then we would have some kind of `Feature` that takes the view and potentially usecases or other callbacks to invoke.

There may be more complicated cases:
* "Reload" is fairly isolated and therefore you were able to implement it here right in `GeckoEngineView`. Other commands like "new tab" will need interaction with other components and even with the app.
* Some commands may need the `EngineView`, but other may not and should even work if there's no browser screen yet, I guess (Think of the home screen and the "new tab" command).

Reply:
I implemented this in a more extensible way.  See https://github.com/gaul/fenix/pull/new/keyboard/basic for an example of some shortcuts.  I am still unclear about the relationship between Android Components, Reference Browser, and Fenix, so please tell me if there is a better approach.

Reply:
I'd like to pick this up, since it seems to be pretty much done. From the discussion above, I don't see a lot of work open on the issue itself, except of fixing the merge conflict.

I locally (haven't pushed it yet), added CTRL+F and CTRL+T, by using these calls:

```
browserToolbarInteractor?.onBrowserToolbarMenuItemTapped(ToolbarMenu.Item.NewTab)
browserToolbarInteractor?.onBrowserToolbarMenuItemTapped(ToolbarMenu.Item.FindInPage)
````

This seems to work as expected.

So what else is to do to get this into a state that can be merged?